### DISCUSSION TITLE
Kubernetes Workload Scaler

### DISCUSSION THREAD

**Discussion by tacf** [kedacore/keda#4393] [General] [answered]
Hi 👋 

I'm currently trying to create a way in order to downscale a deployment as soon as an Argo Rollouts gets created (the purpose of this exercise is to replace one with the other in a hands off approach). I've searched a bit and since we use Keda for our autoscaling i looked into it's scalers and came upon the Workload scaler. Trying to make it work for this situation i came across two issues

1. The scaler does not support a target value of 0 (the ration between the podSelector and the scaled workload)
2. It looks into pod phase only. Which means that even if i set target value to '0.001' (the smallest possible?!) as soon as it sees the pods captured by the selector in 'Running' phase it immediately starts to terminate it's target workload, meaning that we effectively have downtime (it should wait for the 'Ready' status for the running phase?).

My doubt here is, if this is something we could/should implement in this scaler? is it something that suggests a new scaler? or even am i totally off and there's a better way to create this sort of 'if this workload starts up, scale down that other one'

Cheers

**Comment by zroubalik** [comment, marked as answer]
Hi, I think this is a great improvement of the Kubernetes Workload scaler. It definitely could use some improvement. Could you please open new issue with the proposal?

**Reply by tacf** [reply]
For sure, will do.
